google-cloud-storage/src/main/java/com/google/cloud/storage/ObjectReadSessionImpl.java

```diff
@@ -81,54 +77,42 @@ public Object getResource() {
 
   @Override
   public <Projection> Projection readAs(ReadProjectionConfig<Projection> config) {
-    lock.lock();
-    try {
-      checkState(open, "Session already closed");
-      switch (config.getType()) {
-        case STREAM_READ:
-          long readId = state.newReadId();
-          ObjectReadSessionStreamRead<Projection> read =
-              config.cast().newRead(readId, retryContextProvider.create());
-          registerReadInState(readId, read);
-          return read.project();
-        case SESSION_USER:
-          return config.project(this, IOAutoCloseable.noOp());
-        default:
-          throw new IllegalStateException(
-              String.format(
-                  Locale.US,
-                  "Broken java enum %s value=%s",
-                  ProjectionType.class.getName(),
-                  config.getType().name()));
-      }
-    } finally {
-      lock.unlock();
+    checkState(open, "Session already closed");
```

Comment:
locking has been moved into the new ConcurrentIdentityMap, and the try-finally has been removed changing the indentation.